ceno_zkvm/src/scheme/mock_prover.rs

```diff
@@ -392,7 +392,7 @@ impl<'a, E: ExtensionField + Hash> MockProver<E> {
         cb: &CircuitBuilder<E>,
```

Comment:
I suggest in this PR, we can refactor duplicate part from another mock_prover util 
https://github.com/scroll-tech/ceno/blob/cec7b82c29f0d06041bf20aec3e3c0932e6643fa/ceno_zkvm/src/scheme/mock_prover.rs#L748

In summary, in `assert_satisfied_full`
- test on all opcode + table circuit
- output error per opcode/table
- per opcode/table circuit, it only evaluate expression values on errors happened, and just output first row raw value

Since `assert_satisfied_full` is like a super set of ALL opcode circuit of `run_maybe_challenge` for single opcode circuit. So we should extract common part to `run_maybe_challenge`, then `assert_satisfied_full` just call `run_maybe_challenge` per opcode. With that, we remove all the duplicated, while keep a good debugging ability